### ISSUE TITLE
The paths of problems matched by problemmatcher is wrong.

### ISSUE THREAD

**Issue by tinyvan** [microsoft/vscode-cpptools#4836]
![al](https://user-images.githubusercontent.com/27542299/72237906-0c7eb680-3617-11ea-8fe6-7ad8898bded4.png)

**Comment by tinyvan** [comment]
Forgive my poor English....
I am only a ~~junior high school~~ student.
我只是个~~初中~~学生

**Comment by michelleangela** [comment]
@KazDelphi, it looks like the `workspaceFolder` path is duplicated in the path for the file `Government.cpp`. Did you configure any path settings in VS Code or in c_cpp_properties.json? If you made any configurations, can you share those info?

**Comment by tinyvan** [comment]
All configs are default.I didn't change anything.

**Comment by sean-mcmanus** [comment]
This sounds like https://github.com/microsoft/vscode-cpptools/issues/3878 . You need to change your problemMatcher to use an "absolute" or "autoDetect" setting for the "fileLocation".

**Comment by tinyvan** [comment]
This is my `tasks.json`.
```json
{
    "version": "2.0.0",
    "tasks": [
        {
            "type": "shell",
            "label": "clang++ build active file",
            "command": "/usr/bin/clang++",
            "args": [
                "-g",
                "${file}",
                "-o",
                "${fileDirname}/${fileBasenameNoExtension}"
            ],
            "options": {
                "cwd": "/usr/bin"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": "build"
        }
    ]
}
```

**Comment by sean-mcmanus** [comment]
Yes, changing it to 
```
"problemMatcher": {
"base": "$gcc",
"fileLocation": "absolute"
}
```
should fix it (or "autoDetect" instead of "absolute").

**Comment by tinyvan** [comment]
> Yes, changing it to
> 
> ```
> "problemMatcher": {
> "base": "$gcc",
> "fileLocation": "absolute"
> }
> ```
> 
> should fix it (or "autoDetect" instead of "absolute").

Thank you....
Will you fix it in the next version?

**Comment by sean-mcmanus** [comment]
The PR at https://github.com/microsoft/vscode-cpptools/pull/3916 (back in July), fixes it, but it requires a dependency on a newer VS Code version, so the current plan it to ship that fix for 1.1 or so.
